optuna/visualization/_rank.py

```diff
@@ -18,13 +18,20 @@
 from optuna.visualization._utils import _check_plot_args
 from optuna.visualization._utils import _is_log_scale
 from optuna.visualization._utils import _is_numerical
+from optuna.visualization.matplotlib._matplotlib_imports import _imports as matplotlib_imports
 
 
-if _imports.is_successful():
+plotly_is_available = _imports.is_successful()
+if plotly_is_available:
     from optuna.visualization._plotly_imports import go
     from optuna.visualization._plotly_imports import make_subplots
     from optuna.visualization._plotly_imports import plotly
     from optuna.visualization._plotly_imports import Scatter
+if matplotlib_imports.is_successful():
+    # TODO(c-bata): Avoid to import matplotlib here.
```

Comment:
If I understand the context correctly, is it the correct comment? 
```suggestion
    # TODO(c-bata): Import matplotlib visualizer here in case when plotly is not available.
```

Reply:
Thank you for your suggestion!

No, I was suggesting it would be better to avoid import matplotlib. Since the `_get_rank_info` function is intended to share some implementation between plotly and matplotlib visualization functions, so it should not depend on both matplotlib and plotly inside.

Furthermore, I would say todo comments should indicate how we want to change from the current implementation. Perhaps following comment might be more appropriate. What do you think?

```suggestion
    # TODO(c-bata): Refactor to remove matplotlib and plotly dependencies in `_get_rank_info()` for a more modular design.
```

Reply:
Thank you for the update! It looks clearer:)